inst/rmarkdown/templates/copernicus/skeleton/skeleton.Rmd

```diff
@@ -290,10 +290,13 @@ x & y & z\\
 \end{matrix}
 $$
 
-## ALGORITHM
+## ALGORITHM/PROGRAMMING CODE
 
-If you want to use algorithms, you can either enable the required packages in the header (the default, see `algorithms: true`), or make sure yourself that the \LaTeX packages `algorithms` and `algorithmicx` are installed so that `algorithm.sty` respectively `algorithmic.sty` can be loaded by the Copernicus template.
-Copernicus staff will remove all undesirable packages from your LaTeX source code, so please stick to using the header option, which only adds the two acceptable packages.
+If you want to use algorithms or programming code, you can either use the default 'rmarkdown' functionality, which includes the
+code in \LaTeX{} using the `verbatim` environment or using the \LaTeX{} packages `algorithms` and `algorithmicx`. Both need 
+to be available through your preferred \LaTeX{} distribution. 
```

Comment:
So are the two LaTeX package no more accepted by the Copernicus official template and that is why there are removed ?  

Otherwise, the current switch with YAML field is enough, isn't it ? 

I am not quite sure to see the issue here.

Reply:
Yeah, this indeed a little bit confusing. In fact, the two LaTeX packages are still used by Copernicus, and *already* imported through the `copernicus.cls`. However, if we include them in the way we are currently doing it the `*.tex` will come back from the journal's office.

Reply:
I see now. 

I just think the way it is done by the cls file will prevent **tinytex** to install the package automatically if missing. 

So algorithm won't work in the skeleton unless installed manually by the user. This goes a bit against the use of **rmarkdown** + **tinytex** to compile a document without worrying about CTAN packages. 

by adding the `\usepackage` in template this was triggering the installation if missing. 

Should be install those packages in `copernicus_article` because they are needed anyway by the cls file to use all feature ? 

Or maybe in the setup chunk of the skeleton ? 

I think we have already done that with a template for the same reason: a cls file while loading a package if already installed. But this means that if not, **tinytex** won't see any error to trigger an installation. 

https://github.com/rstudio/rticles/blob/10d269397fe3af52ffeebfab533a3eb970567f33/inst/rmarkdown/templates/jasa/skeleton/skeleton.Rmd#L59-L61

Reply:
Do you suggest I should implement this in the skeleton for all packages? Might be a solution indeed.

Reply:
Ah, strike my comment above :-) If it's alright to do this in the skeleton, I think that is a good idea - it won't be needed for users without algorithms, but it won't really hurt them either.

Reply:
I think this could be put in the skeleton in the setup chunk, at least for this two packages that we know needs to be install for the skeleton to work. 

We could indeed add this for all packages in your list
```r
if (tinytex::is_tinytex()) {
	local({
		  required <- c(...)
		  installed <- tinytex::check_installed(required)
		  tinytex::tlmgr_install(required[!installed])
	})
}
```

One could remove it if it is not desired.

Reply:
Stupid question: Is there any argument against implementing this directly in the `copernicus_article.R`? If there is, I would add it to the `*.Rmd` file instead as suggested.